### ISSUE TITLE
zip_open(), incorrect return type

### ISSUE THREAD

**Issue by ArtemGoutsoul** [vimeo/psalm#3029]
https://psalm.dev/r/f6e847c5a3
```
/**
 * @param string $filename
 * @return resource
 **/
function open_zip(string $filename)
{
	$r = zip_open($filename);
	if (is_resource($r)) return $r;
    throw new Exception("Fail!");
}
```
ERROR: RedundantCondition - 10:6 - Found a redundant condition when evaluating $r and trying to reconcile type 'resource' to resource

This is incorrect since zip_open can return a number:
"Returns a resource handle for later use with zip_read() and zip_close() or returns the number of error if filename does not exist or in case of other error."

See https://www.php.net/manual/en/function.zip-open.php

**Comment by muglug** [comment]
Please open a PR in `CallMap.php`!

**Comment by LeSuisse** [comment]
This one can be closed, fixed in #3046
